Simplenote/src/main/java/com/automattic/simplenote/NoteEditorFragment.java

```diff
@@ -434,9 +435,7 @@ public void run() {
     @Override
     public void onPause() {
         super.onPause();  // Always call the superclass method first
-
```

Comment:
how would this have been different/broken if we had moved `saveNote()` up here instead of moving `mNotesBucket.stop()` to `onDestroy()`?

Reply:
> I'm guessing the mNote.save(); statement in the saveNote() method doesn't save/sync immediately and queues the changes for later.

That should be true, and also it only sends one change at a time and waits for each to return before applying the next ones - it may have to rewrite the internal changes.

That is, I can see where we definitely want to avoid this…

```java
saveNote();
mNotesBucket.stop();
```

…because that's going to interrupt a process that's relatively slow (makes and waits for a network call) _and_ asynchronous.

Questioning as below, do we know that we need to stop the bucket connection on pause? According to [the `onPause` docs](https://developer.android.com/guide/components/activities/activity-lifecycle) it looks like they are suggesting that this kind of thing be done in `onSave()`

It's easy to see both arguments for stopping the bucket connection: stop sending and listening to network calls when the app's not in the foreground; keep sending queued changes and listening for new ones so that when the app returns it will be ready.

Reply:
> According to [the `onPause` docs](https://developer.android.com/guide/components/activities/activity-lifecycle) it looks like they are suggesting that this kind of thing be done in `onSave()`

Do you mean `onStop` rather than `onSave` here?  If so, I think that's being discuss in [the other comment thread](https://github.com/Automattic/simplenote-android/pull/880#discussion_r345260215).